From 6dfac7d2772b56545e045c325564fc28a0e401cb Mon Sep 17 00:00:00 2001
From: Gaurav Jain <gaurav.jain@nxp.com>
Date: Thu, 30 Nov 2023 14:33:25 +0530
Subject: [PATCH] Fix build for Linux 6.7

crypto_ahash_alignmask() is removed in kernel,
as it always return 0 and nonzero alignmask are not
supported anymore.

see also: https://git.kernel.org/pub/scm/linux/kernel/git/torvalds/linux.git/commit/?id=0f8660c82b79af595b056f6b9f4f227edeb88574

Upstream-Status: Pending
Signed-off-by: Gaurav Jain <gaurav.jain@nxp.com>
---
 cryptlib.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/cryptlib.c b/cryptlib.c
index 4d739e5..ef53a6e 100644
--- a/cryptlib.c
+++ b/cryptlib.c
@@ -381,7 +381,9 @@ int cryptodev_hash_init(struct hash_data *hdata, const char *alg_name,
 	}
 
 	hdata->digestsize = crypto_ahash_digestsize(hdata->async.s);
+#if (LINUX_VERSION_CODE < KERNEL_VERSION(6, 7, 0))
 	hdata->alignmask = crypto_ahash_alignmask(hdata->async.s);
+#endif
 
 	init_completion(&hdata->async.result.completion);
 
-- 
2.25.1

